operatorcert/entrypoints/create_container_image.py

```diff
@@ -0,0 +1,141 @@
+import argparse
+import logging
+from typing import Any
+from urllib.parse import urljoin
+
+from operatorcert import pyxis
+from operatorcert.logger import setup_logger
+
+LOGGER = logging.getLogger("operator-cert")
+
+
+def setup_argparser() -> Any:
+    """
+    Setup argument parser
+
+    Returns:
+        Any: Initialized argument parser
+    """
+    parser = argparse.ArgumentParser(description="Bundle dockerfile generator.")
+
+    parser.add_argument(
+        "--pyxis-url",
+        default="https://catalog.redhat.com/api/containers/",
+        help="Base URL for Pyxis container metadata API",
+    )
+    parser.add_argument(
+        "--isv-pid",
+        help="isv_pid of the certification project from Red Hat Connect",
+        required=True,
+    )
+    parser.add_argument(
+        "--repo-published",
+        help="is the ContainerImage repository published?",
+        required=True,
+    )
+    parser.add_argument(
+        "--registry", help="Certification Project Registry", required=True
+    )
+    # TODO
+    parser.add_argument("--repository", help="", required=True)
+
+    parser.add_argument(
+        "--certified", help="Is the ContainerImage certified?", required=True
+    )
+    parser.add_argument(
+        "--docker-image-digest",
+        help="Container Image digest of related Imagestream",
+        required=True,
+    )
+    parser.add_argument("--verbose", action="store_true", help="Verbose output")
+    return parser
+
+
+def check_if_image_already_exists(args) -> bool:
+    check_url = urljoin(
+        args.pyxis_url,
+        f"v1/images?page_size=1&"
+        f"filter="
+        f"isv_pid=={args.isv_pid};"
+        f"docker_image_digest=={args.docker_image_digest}",
+    )
+
+    # Get the list of the ContainerImages with given parameters
+    rsp = pyxis.get(check_url)
+    rsp.raise_for_status()
+
+    query_results = rsp.json()["data"]
+    total_images = rsp.json()["total"]
+
+    if len(query_results) == 0:
+        return False
+
+    # Test, if the images that we got are all deleted.
+    # To do that, we firstly check if record in response is deleted.
+    # If it is not, image already exists. If it is, then we query
+    # all the deleted images with given parameters,
+    # and compare amount with the original amount.
+    # That's most efficient method, since we cannot use `$exists` in Pyxis query.
+    if query_results[0].get("deleted", False):
```

Comment:
I'm a bit confused by this. Why isn't it sufficient to go through the list of query_results and check the `deleted` field of each, and just return `True` if any is not deleted, `False` otherwise?

Reply:
It's because of the border case when there will be more deleted containers that will fit on one page. 
Also, a bigger page size (currently it's set to 1) would cost more resources

Reply:
I am trying to understand the logic of this function and I can't figure out why it is so complicated. If my assumption is correct I think you can use a `filter` like this `filter=not(deleted==true)` to get information about whether the image exists and hasn't been deleted. 

If you are afraid of cost of resources you should start using `include` argument to fetch a minimal set of required fields that you need.

Reply:
You cannot do filter=not(deleted==true), because it won't work if field `deleted` doesn't exist- and usually, it doesn't exist.

Reply:
oh wait, you are right, I can do that. I tried that before and had some issues with it.